### DISCUSSION TITLE
Markdown Shortcode Features

### DISCUSSION THREAD

**Discussion by isaaclombardssw** [tinacms/tinacms#4794] [Experimental]
Tina comes inbuilt with a few ways to customize the rich-text fields of the schema. 

One of these is with shortcode – in raw markdown, templates can be encoded with some custom opening and closing tags with the `match` fields.

While this feature has been around for some time, it's still experimental and has a few caveats to its usage.

This discussion is about suggestions or fixes to the markdown shortcode features!

**Comment by niccokunzmann** [comment]
Hello, I am very happy that the shortcodes feature exists.
I just copied in the example and every time I try to embed the shortcode, the page closes.

![grafik](https://github.com/user-attachments/assets/8db07943-c284-4034-a1cd-155579c00b1e)

And, I wonder which 'react components' I can actually choose... Is there a list?

```
            ui: {
              component: 'textarea',
            },
```

![grafik](https://github.com/user-attachments/assets/e9465557-6ac1-473a-8b50-70e35b99564d)
